tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -54,30 +55,42 @@ static Startup()
             if (automaticTraceEnabled || customTracingEnabled || needsDogStatsD)
             {
                 StartupLogger.Log("Invoking managed method to start external processes.");
-                TryInvokeManagedMethod("Datadog.Trace.AgentProcessManager", "Initialize");
+                TryInvokeManagedMethod("Datadog.Trace.AgentProcessManager", "Initialize", "Datadog.Trace.AgentProcessManagerLoader");
             }
 
             if (automaticTraceEnabled)
             {
                 StartupLogger.Log("Invoking managed tracer.");
-                TryInvokeManagedMethod("Datadog.Trace.ClrProfiler.Instrumentation", "Initialize");
+                TryInvokeManagedMethod("Datadog.Trace.ClrProfiler.Instrumentation", "Initialize", "Datadog.Trace.ClrProfiler.InstrumentationLoader");
             }
         }
 
         internal static string ManagedProfilerDirectory { get; }
 
-        private static void TryInvokeManagedMethod(string typeName, string methodName)
+        private static void TryInvokeManagedMethod(string typeName, string methodName, string? loaderHelperTypeName = null)
         {
             try
             {
                 var assembly = LoadAssembly(AssemblyName);
-
                 if (assembly == null)
                 {
                     StartupLogger.Log("Assembly '{0}' cannot be loaded. The managed method ({1}.{2}) cannot be invoked", AssemblyName, typeName, methodName);
                     return;
                 }
 
+                if (loaderHelperTypeName is not null)
+                {
+                    // The loader helper type name is a class that calls the initialization in the .ctor
+                    // this way we avoid the reflection invoke call.
+                    if (assembly.GetType(loaderHelperTypeName, throwOnError: false) is { } loaderHelperType)
+                    {
+                        Activator.CreateInstance(loaderHelperType);
```

Comment:
Are you sure `Activator.CreateInstance` is faster than invoking a method through reflection? 🤔

Reply:
I'm trusting @andrewlock on this one: https://andrewlock.net/benchmarking-4-reflection-methods-for-calling-a-constructor-in-dotnet/ 🙂🙂🙂🙂

Reply:
Is not exactly the same case, but very similar, and also one thing to take in consideration is that on startup time in the tracer we have NGEN disabled and instrumenting through the JITCompilationStart to inject the loader. So less internal calls the better. And I think calling by reflection will result in more checks and internals calls, althought I'm not 100% sure of this.

Reply:
```
BenchmarkDotNet=v0.12.1, OS=Windows 10.0.22000
AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
.NET Core SDK=7.0.100-preview.7.22377.5
  [Host]     : .NET Core 3.1.28 (CoreCLR 4.700.22.36202, CoreFX 4.700.22.36301), X64 RyuJIT
  DefaultJob : .NET Core 3.1.28 (CoreCLR 4.700.22.36202, CoreFX 4.700.22.36301), X64 RyuJIT


|                    Method |     Mean |    Error |   StdDev |   Median |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------------- |---------:|---------:|---------:|---------:|-------:|------:|------:|----------:|
| ReflectionInvokeBenchmark | 74.25 ns | 0.646 ns | 0.605 ns | 74.37 ns |      - |     - |     - |         - |
|        ActivatorBenchmark | 30.30 ns | 0.760 ns | 2.194 ns | 30.90 ns | 0.0000 |     - |     - |      24 B |
```

```BenchmarkDotNet=v0.12.1, OS=Windows 10.0.22000
AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
  [Host]     : .NET Framework 4.8 (4.8.4515.0), X64 RyuJIT
  DefaultJob : .NET Framework 4.8 (4.8.4515.0), X64 RyuJIT


|                    Method |      Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------------- |----------:|---------:|---------:|-------:|------:|------:|----------:|
| ReflectionInvokeBenchmark | 103.12 ns | 0.835 ns | 0.740 ns |      - |     - |     - |         - |
|        ActivatorBenchmark |  56.75 ns | 0.299 ns | 0.280 ns | 0.0038 |     - |     - |      24 B |
```